client/components/jetpack/wpcom-business-at/index.tsx

```diff
@@ -91,25 +102,78 @@ const contentPerPrimaryProduct = {
 	},
 };
 
+interface BlockingHoldNoticeProps extends Props {
+	siteId: number;
+}
+
+function BlockingHoldNotice( { siteId, product }: BlockingHoldNoticeProps ): ReactElement | null {
+	const content = React.useMemo( () => contentPerPrimaryProduct[ product ], [ product ] );
+	const { eligibilityHolds: holds } = useSelector( ( state ) => getEligibility( state, siteId ) );
+	if ( ! holds ) {
+		return null;
+	}
+
+	const blockingMessages = getBlockingMessages( translate );
+	blockingMessages.BLOCKED_ATOMIC_TRANSFER.message = translate(
+		'This site is currently not eligible for %s. Please contact our support team for help.',
+		{ args: [ content.header ] }
+	);
+
+	return (
+		<HardBlockingNotice
+			translate={ translate }
+			holds={ holds }
+			blockingMessages={ blockingMessages }
+		/>
+	);
+}
+
 export default function WPCOMBusinessAT( { product }: Props ): ReactElement {
 	const content = React.useMemo( () => contentPerPrimaryProduct[ product ], [ product ] );
-	const siteId = useSelector( getSelectedSiteId );
+	const siteId = useSelector( getSelectedSiteId ) as number;
+
+	// Gets the site eligibility data.
+	const isEligible = useSelector( ( state ) => isEligibleForAutomatedTransfer( state, siteId ) );
+	const {
+		eligibilityHolds: holds,
+		eligibilityWarnings: warnings,
+	}: EligibilityData = useSelector( ( state ) => getEligibility( state, siteId ) );
+	const automatedTransferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, siteId )
+	);
+
+	// Check if there's a blocking hold.
+	const hasBlockingHold = ! isEligible || ( holds && hasBlockingHoldFunc( holds ) );
+
+	// Gets state to control dialog and continue button.
+	const [ showDialog, setShowDialog ] = useState( false );
+	const [ approvedTransfer, setApproveTransfer ] = useState(
+		0 === warnings?.length && 0 === holds?.length
+	);
+	const onClose = () => setShowDialog( false );
+
+	// Handles dispatching automated transfer.
 	const dispatch = useDispatch();
 	const initiateAT = React.useCallback( () => {
-		dispatch( initiateThemeTransfer( siteId, null, null ) );
-	}, [ dispatch, siteId ] );
+		if ( approvedTransfer ) {
+			dispatch( initiateThemeTransfer( siteId, null, '' ) );
+		} else {
+			setShowDialog( true );
+		}
+	}, [ dispatch, siteId, approvedTransfer ] );
 	const eventName =
 		product === 'backup'
 			? 'calypso_jetpack_backup_business_at'
 			: 'calypso_jetpack_scan_business_at';
 	const trackInitiateAT = useTrackCallback( initiateAT, eventName );
-
-	const automatedTransferStatus = useSelector( ( state ) =>
-		getAutomatedTransferStatus( state, siteId )
-	);
+	const approveAndInitiateAt = () => {
+		setApproveTransfer( true );
+		trackInitiateAT();
```

Comment:
I think we need to call `onClose` here, otherwise the modal remains open when you confirm the AT.